sqlite/src/vdbeaux.c

```diff
@@ -4848,7 +4875,51 @@ int sqlite3MemCompare(const Mem *pMem1, const Mem *pMem2, const CollSeq *pColl){
 
 #if defined(SQLITE_BUILDING_FOR_COMDB2)
   if(combined_flags&MEM_Datetime || combined_flags&MEM_Interval) {
-    return compareDateTimeInterval(f1, f2, combined_flags, pMem1, pMem2);
+    if( f1&(MEM_Int|MEM_Real) ){
+      /* If lhs is a number, convert rhs to a real and compare these 2 numbers.
+         This takes care of out-of-range values. */
+      sqlite3VdbeMemInit(&copy, pMem1->db, MEM_Null);
+      sqlite3VdbeMemCopy(&copy, pMem2);
+      sqlite3VdbeMemRealify(&copy);
+
+      f2 = copy.flags;
+      pMem2 = &copy;
+      combined_flags = f1|f2;
+    }else if( f2&(MEM_Int|MEM_Real) ){
+      /* If rhs is a number, convert lhs to a real and compare these 2 numbers.
+         This takes care of out-of-range values. */
+      sqlite3VdbeMemInit(&copy, pMem2->db, MEM_Null);
+      sqlite3VdbeMemCopy(&copy, pMem1);
+      sqlite3VdbeMemRealify(&copy);
+
+      f1 = copy.flags;
+      pMem1 = &copy;
+      combined_flags = f1|f2;
+    }else{
+      u8 err;
+      int rc = compareDateTimeInterval(f1, f2, combined_flags, pMem1, pMem2, &err);
+      if( err==0 )
+        return rc;
+      /* Type error. If one value is a string, convert the other one to
+         a string and then compare. */
+      if( f1&MEM_Str ){
+        sqlite3VdbeMemInit(&copy, pMem1->db, MEM_Null);
+        sqlite3VdbeMemCopy(&copy, pMem2);
+        sqlite3VdbeMemStringify(&copy, pColl ? pColl->enc : pMem1->enc, 0);
+
```

Comment:
What if this call to sqlite3VdbeMemStringify fails?

Reply:
I added a check for the return code. Thanks for pointing it out!